cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,916 @@
+        allocate_dataframe_buffer<thrust::tuple<vertex_t, vertex_t>>(intersection_indices.size(),
+                                                                     handle.get_stream());
+
+      thrust::tabulate(
+        handle.get_thrust_policy(),
+        get_dataframe_buffer_begin(vertex_pair_buffer_q_r_edge_p_q),
+        get_dataframe_buffer_end(vertex_pair_buffer_q_r_edge_p_q),
+        generate_p_r_q_r<vertex_t, edge_t, false>{
+          raft::device_span<size_t const>(intersection_offsets.data(), intersection_offsets.size()),
+          raft::device_span<vertex_t const>(intersection_indices.data(),
+                                            intersection_indices.size()),
+          raft::device_span<vertex_t>(edgelist_srcs.data() + num_valid_edges, num_invalid_edges),
+          raft::device_span<vertex_t>(edgelist_dsts.data() + num_valid_edges, num_invalid_edges)});
+
+      // Unrolling the edges require the edges to be sorted by destination
+      // re-sort the invalid edges by 'dst'
+      thrust::sort_by_key(handle.get_thrust_policy(),
+                          transposed_edge_first + num_valid_edges,
+                          transposed_edge_first + edgelist_srcs.size(),
+                          num_triangles.begin() + num_valid_edges);
+
+      thrust::for_each(handle.get_thrust_policy(),
+                       thrust::make_counting_iterator<edge_t>(0),
+                       thrust::make_counting_iterator<edge_t>(intersection_indices.size()),
+                       unroll_edge<vertex_t, edge_t, decltype(transposed_edge_first)>{
+                         edge_t{num_valid_edges},
+                         raft::device_span<edge_t>(num_triangles.data(), num_triangles.size()),
+                         get_dataframe_buffer_begin(vertex_pair_buffer_p_r_edge_p_q),
+                         transposed_edge_first,
+                         transposed_edge_first + num_valid_edges,
+                         transposed_edge_first + edgelist_srcs.size()});
+
+      thrust::for_each(handle.get_thrust_policy(),
+                       thrust::make_counting_iterator<edge_t>(0),
+                       thrust::make_counting_iterator<edge_t>(intersection_indices.size()),
+                       unroll_edge<vertex_t, edge_t, decltype(transposed_edge_first)>{
+                         edge_t{num_valid_edges},
+                         raft::device_span<edge_t>(num_triangles.data(), num_triangles.size()),
+                         get_dataframe_buffer_begin(vertex_pair_buffer_q_r_edge_p_q),
+                         transposed_edge_first,
+                         transposed_edge_first + num_valid_edges,
+                         transposed_edge_first + edgelist_srcs.size()});
+
+      // case 2: unroll (q, r)
+      // For each (q, r) edges to unroll, find the incoming edges to 'r' let's say from 'p' and
+      // create the pair (p, q)
+      cugraph::find_unroll_p_r_or_q_r_edges<vertex_t, edge_t, false, true>(
```

Comment:
I think "find_" here is a bit confusing. Just `unroll_p_r_or_q_r_edges` is clearer.

Reply:
I added find because I needed the to find the p, r and q, r edges first before unrolling them. But I renamed it as you suggested.